db_test.go

```diff
@@ -145,14 +145,19 @@ func TestSaveExistingDocument(t *testing.T) {
 
 		doc := c.NewDocument()
 		doc.Set(key, "clover")
-		id, err := db.InsertOne(collection, doc)
+		err = db.Save(collection, doc)
 		require.NoError(t, err)
 
+		n, err := db.Query(collection).Count()
```

Comment:
After using `Save()` to insert the document at line 148, retrieve it and test if it is equal to the one you just created.
```go
savedDoc, err := db.Query("todos").FindFist()
require.NoError(t, err)
require.Equal(savedDoc)
```